torch/csrc/DynamicTypes.cpp

```diff
@@ -82,9 +82,13 @@ PyObject* createPyObject(const at::Storage& storage) {
       // This is useful for checking aliasing info from python
       dynamic_cast<at::functionalization::FunctionalStorageImpl*>(
           storage.unsafeGetStorageImpl()) == nullptr) {
-    TORCH_CHECK_NOT_IMPLEMENTED(
-        false,
-        "python bindings to nullptr storage (e.g., from torch.Tensor._make_wrapper_subclass) are currently unsafe and thus disabled.  See https://github.com/pytorch/pytorch/issues/61669 for more details");
+    // TODO: allowing wrapper tensor subclasses to call .storage()
+    // is dangerous, but needed for AOTAutograd analysis.
+    // Some options:
+    // (1) remove the check
+    // (2) introduce ReadOnlyStorage
+    // (3) Add private TLS telling us when it is ok to return a null storage,
+    //     use it in AOTAutograd
```

Comment:
Left this comment for PR review. I think that (3) would be pretty easy to add, if we think that removing the check on .storage() calls is too much of a footgun.

Reply:
So, at a high level, I'm not sure how you expect this to work. Let's imagine I have a tensor subclass, which under the hood is "just" a dense tensor (but it has some extra metadata and non-standard semantics for operations). In that case, I had better report the underlying storage, because I need to faithfully replicate the aliasing semantics so that AOTAutograd does the right thing.) How is what you did here going to do the right thing?